--------------------------------------------------------------------------------
loop.try
--------------------------------------------------------------------------------
loop.try
loop.catch AnException
main.catch Exception
